### PR TITLE
feat: Add Location,Type, time filters

### LINKED ISSUE #1672: Add Location , time , type filter Like Event Brite
**Describe the feature you'd like**
EventBrite has Location , Time & Type filter to for filtering event

**Screenshot**
photo6201859894580193446

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by theLimitBreaker**
Fixes #1672 

Screenshots for the change:

### CODE REVIEW

**Comment by theLimitBreaker** [comment]
@angmas1 @ShridharGoel  Peer review please

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt [hunk 1]
```diff
@@ -1,39 +1,44 @@
...
 import org.fossasia.openevent.general.utils.Utils.setToolbar
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class SearchFilterFragment : Fragment() {
 
     private lateinit var rootView: View
     private var isFreeStuffChecked = false
+    private var selectedTime: String? = null
+    private var selectedLocation: String? = null
+    private var selectedCategory: String? = null
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt]: why not `lateinit` ?

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt [hunk 2]
```diff
@@ -79,7 +73,60 @@ class SearchFilterFragment : Fragment() {
...
+                    Navigation.findNavController(rootView)
+                        .navigate(R.id.searchTimeFragment, it, getAnimFade())
+                }
+        }
+
+        rootView.tvSelectLocation.text = selectedLocation
+        rootView.tvSelectLocation.setOnClickListener {
+            SearchTimeFragmentArgs(
+                time = selectedLocation,
+                fromFragmentName = "SearchFilterFragment",
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt]: use a constant. elsewhere too

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt (hunk 1 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt]: I read somewhere along issues where areebJamal said "use lateinit as last resort". No other specific reason.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt (hunk 2 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFilterFragment.kt]: done

**Comment by theLimitBreaker** [comment]
@nikit19  updated please review

**Comment by nikit19** [comment]
@theLimitBreaker please resolve conflicts

**Comment by theLimitBreaker** [comment]
@nikit19  @iamareebjamal  updated

**Pull request merged**